numpy/random/tests/test_random.py

```diff
@@ -209,6 +243,38 @@ def test_repeatability(self):
         res = hashlib.md5(val).hexdigest()
         assert_(tgt[np.dtype(np.bool).name] == res)
 
+    def test_int64_uint64_corner_case(self):
+        # When stored in Numpy arrays, `lbnd` is casted
+        # as np.int64, and `ubnd` is casted as np.uint64.
+        # Checking whether `lbnd` >= `ubnd` used to be
+        # done solely via direct comparison, which is incorrect
+        # because when Numpy tries to compare both numbers,
+        # it casts both to np.float64 because there is
+        # no integer superset of np.int64 and np.uint64. However,
+        # `ubnd` is too large to be represented in np.float64,
+        # causing it be round down to np.iinfo(np.int64).max,
+        # leading to a ValueError because `lbnd` now equals
+        # the new `ubnd`.
+
+        dt = np.int64
+        tgt = np.iinfo(np.int64).max
+        lbnd = np.int64(np.iinfo(np.int64).max)
+        ubnd = np.uint64(np.iinfo(np.int64).max + 1)
+
+        # None of these function calls should
+        # generate a ValueError now
+        actual = np.random.randint(lbnd, ubnd, dtype=dt)
+        assert_equal(actual, tgt)
+
+        actual = np.random.randint([lbnd] * 3, ubnd, dtype=dt)
+        assert_equal(actual, tgt)
+
+        actual = np.random.randint(lbnd, [ubnd] * 3, dtype=dt)
+        assert_equal(actual, tgt)
+
+        actual = np.random.randint([lbnd] * 3, [ubnd] * 3, dtype=dt)
```

Comment:
I strongly disagree with what you are saying here.  First of all, `randint` by design is supposed to not include the endpoint.  That's not going to change here, and we in fact deprecated the method where the endpoint was included, so already, your suggestion for including closed intervals is something I would not really count on getting much support for.

Also, why should we complicate the API even further by allowing wrapping?  That's only going to cause more confusion and make the function harder to maintain.  You can already see just how complicated it is to maintain as is.

Finally, I don't understand why you think those tests should error.  Those are perfectly valid inputs that fit into the range of `np.int64`.  I don't really see why you think those should error.

In general, I'm finding that many of your comments regarding the design are just assertions without much justification as to why you think that is the case.  For someone who has worked on this PR extensively and discussed this with multiple maintainers, I'm having a hard time following your train of thought.